packages/kit/src/packaging/index.js

```diff
@@ -104,6 +104,27 @@ export async function make_package(config, cwd = process.cwd()) {
 	}
 
 	pkg.exports = { ...generated, ...pkg.exports };
+
+	// Several heuristics in Kit/vite-plugin-svelte to tell Vite to mark Svelte packages
+	// rely on the "svelte" property. Vite/Rollup/Webpack plugin can all deal with it.
+	// See https://github.com/sveltejs/kit/issues/1959 for more info and related threads.
+	if (pkg.exports['.']) {
+		const svelte_export = typeof pkg.exports['.'] === 'string' ? pkg.exports['.'] : undefined;
+		if (svelte_export) {
+			pkg.svelte = svelte_export;
+		} else {
+			console.warn(
+				'The "." entry in "exports" is not a string. ' +
+					'If you set it by hand, please also set one of the options as a "svelte" entry point'
+			);
+		}
+	} else {
+		console.warn(
+			'The "." entry in "exports" is not a string. ' +
+				'If you set it by hand, please also set one of the options as a "svelte" entry point'
```

Comment:
We might want to throw an error for this one

```suggestion
			'The "." entry in "exports" does not exists. ' +
				'Please specify one or set a "svelte" entry point yourself'
```

Reply:
Why error in this case? It could be 

```
"." {  "import": "./index.js" }
```

Reply:
Looks like it isn't that clear from this comment view, but this is for the outer `else` block where there's no `"."` export as well. Also, yes, never mind about throwing the error.